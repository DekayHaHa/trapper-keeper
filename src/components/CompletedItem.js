import React from 'react'

export const CompletedItem = (props) => {
	return (
		<div>
			<p className='uncompleted-item'>{props.text}</p><button onClick={() => props.toggle(props.id)}>Check</button>
		</div>
	)
}