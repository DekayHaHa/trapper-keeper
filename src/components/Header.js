import React from 'react';
import CreateNote from '../containers/CreateNote';
import { Link } from 'react-router-dom';

export const Header = () => {
  return (
    <header className="header-container">
      <Link to="/" className='link-style'>
        <h1>Task Mancer</h1>
      </Link>
      {/* <h1>Task Mancer</h1> */}
      <Link to="/api/new-note" className="new-note-btn">
        <CreateNote />
      </Link>
    </header>
  );
};
