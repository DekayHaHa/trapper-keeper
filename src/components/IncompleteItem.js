import React from 'react'

export const IncompleteItem = (props) => {
	return (
		<div>
			<p className='completed-item'>{props.text}</p><button onClick={() => props.toggle(props.id)}>Uncheck</button>
		</div>
	)
}