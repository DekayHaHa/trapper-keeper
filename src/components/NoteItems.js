import React from 'react';
import { Divider, Typography } from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles';

export const NoteItems = ({ noteItems, classes }) => {
    const itemsList = noteItems.map(item => (
      <div iscomplete={JSON.stringify(item.isComplete)} key={item.id} >
        <Typography id={item.id} className={item.isComplete ? classes.strikethrough : ''} style={{ color: '#000' }} >
          {item.text}
        </Typography>
        <Divider light />
      </div>
    ));

  console.log('items list: ', itemsList)
  console.log(classes)
    return (
      <div>
        {itemsList.filter(item => item.props.iscomplete === 'false')}
        <Divider className={classes.divider} />
        {itemsList.filter(item => item.props.iscomplete === 'true')}
      </div>
    );
};

const styles = {
  divider: {
    backgroundColor: '#9445FF'
  },
  strikethrough: {
    textDecoration: 'line-through'
  }
};

export default withStyles(styles)(NoteItems);