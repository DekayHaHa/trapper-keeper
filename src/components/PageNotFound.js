import React from 'react';
import { Link } from 'react-router-dom';

export const PageNotFound = () => {
	return (
		<div className='page-not-found'>
			<h2>This is not the page</h2>
			<h2>you are looking for.</h2>
			<h2 className='not-found'>404</h2>
			<Link to={'/'} className='link'>Return Home</Link>
		</div>
	)
}