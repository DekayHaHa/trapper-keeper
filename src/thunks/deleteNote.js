import { hasError, removeNote } from '../actions';

export const deleteNote = (id) => {
  const option = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const url = `http://localhost:3001/api/notes/${id}`;
  return async (dispatch) => {
    try {
      const response = await fetch(url, option);
      if(!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(removeNote(id))
    } catch(error) {
      dispatch(hasError(error.message));
    }
  }
}