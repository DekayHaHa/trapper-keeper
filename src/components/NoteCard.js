import React, {Component} from 'react';
import { connect } from 'react-redux';
import { Card, Tooltip, IconButton, CardContent, CardHeader } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';
import { NoteItems } from './NoteItems';
import Icon from '@material-ui/core/Icon';
import { deleteNote } from '../thunks/deleteNote';

class NoteCard extends Component {

    render() {
        const { note, classes, deleteNote } = this.props;
        console.log('note card props: ', this.props.note)
      return (
        <Tooltip title='Edit Note' placement='bottom' enterDelay={500}>
            <Card className={classes.card}>
                <Link to='/' className={classes.delete}>
                    <Tooltip title='Delete Note'>
                    <IconButton onClick={() => deleteNote(note.id)}>
                        <Icon>delete</Icon>
                    </IconButton>
                    </Tooltip>
                </Link>
                <Link to={`/api/notes/${note.id}`} className={classes.link}>
                    <CardHeader title={note.title} classes={{ title: classes.title }} />
                    <CardContent>
                        <ul>
                            <NoteItems noteItems={note.itemsList} />
                        </ul>
                    </CardContent>
                  </Link>
            </Card>
        </Tooltip>
      )
    }
}

const styles = {
    link: {
        textDecoration: 'none'
    },
    card: {
        margin: 12
    },
    title: {
        color: '#515151'
    },
    delete: {
        float: 'right',
        margin: '5px'
    }
};

export const mapDispatchToProps = (dispatch) => ({
	deleteNote: (id) => dispatch(deleteNote(id))
});

export default withStyles(styles)(connect(null, mapDispatchToProps)(NoteCard));
