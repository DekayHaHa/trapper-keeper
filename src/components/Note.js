import React from 'react';
import { NoteCard } from '../containers/NoteCard';
import { Grid } from '@material-ui/core';

export const Note = (props) => {
  // const renderNoteItems = props.itemsList.map(note => (
  //   <Grid key={Date.now()} item xs={12} sm={6} md={4} lg={2}>
  //     <NoteCard key={note.id} note={note} />
  //   </Grid>
  // ));

  const renderNoteItems = () => {
    props.itemsList.map(item => console.log('note item: ', item));
    // return props.itemsList.map(item => <li>{item.text}</li>)
  }

  console.log('props in note: ', props.itemsList)

  return (
    // <Grid container justify="flex-start" alignItems="center">
    //   {renderNoteItems}
    // </Grid>
    <div>
      {renderNoteItems}
    </div>
  );
};