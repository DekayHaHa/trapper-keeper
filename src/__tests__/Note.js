import React from 'react';
import { Note, mapDispatchToProps } from '../components/Note';
import { shallow } from 'enzyme';
import { deleteNote } from '../thunks/deleteNote';

jest.mock('../thunks/deleteNote');

describe('Note', () => {
  let mockItemsList;
  let wrapper;
  let mockFn;

  beforeEach(() => {
    mockFn = jest.fn();
    mockItemsList = [{text: "item", isComplete: false, id: 1554329952339}]
    wrapper = shallow(<Note title={'title'} itemsList={mockItemsList} deleteNote={mockFn}/>);
  });

  it('should match snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke deleteNote on removeNote', () => {
    wrapper.instance().removeNote();
    expect(mockFn).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a deleteNote action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = deleteNote(2);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.deleteNote(2);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  });
});