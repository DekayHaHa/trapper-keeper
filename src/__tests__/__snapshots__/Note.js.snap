// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note should match snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Note
    deleteNote={[MockFunction]}
    itemsList={
      Array [
        Object {
          "id": 1554329952339,
          "isComplete": false,
          "text": "item",
        },
      ]
    }
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          title
        </p>,
        <ul>
          <li>
            item
          </li>
        </ul>,
        <button
          onClick={[Function]}
        >
          Delete Note
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
        },
        "ref": null,
        "rendered": "title",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              item
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1554329952339",
            "nodeType": "host",
            "props": Object {
              "children": "item",
            },
            "ref": null,
            "rendered": "item",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete Note",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete Note",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            title
          </p>,
          <ul>
            <li>
              item
            </li>
          </ul>,
          <button
            onClick={[Function]}
          >
            Delete Note
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
          },
          "ref": null,
          "rendered": "title",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                item
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1554329952339",
              "nodeType": "host",
              "props": Object {
                "children": "item",
              },
              "ref": null,
              "rendered": "item",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete Note",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete Note",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
